src/app/courses/courses-list/course-item/course-item.component.spec.ts

```diff
@@ -32,13 +34,15 @@ describe('CourseItemComponent', () => {
   it('it should update correctly when input property changes', () => {
     const inputValue = 'test description';
     component.description = inputValue;
-    fixture.detectChanges();
+    const changeDetectorRef = fixture.debugElement.injector.get<ChangeDetectorRef>(ChangeDetectorRef);
+    changeDetectorRef.detectChanges();
     const descriptionElement = fixture.debugElement.query(By.css('.courseItemDescription')).nativeElement;
     expect(descriptionElement.textContent).toEqual(inputValue);
 
     const newInputValue = 'new test description';
     component.description = newInputValue;
-    fixture.detectChanges();
+    const changeDetectorRef2 = fixture.debugElement.injector.get<ChangeDetectorRef>(ChangeDetectorRef);
+    changeDetectorRef2.detectChanges();
```

Comment:
Actually calling the `fixture.detectChanges()`  achieves the same result as calling `changeDetectorRef.detectChanges()`. It triggers change detection for the component and its template

Reply:
If I use `fixture.detectChanges()` it doesn't rerender component during tests. I got error  `Expected '' to equal 'test description'`

Reply:
yes, it works, I have redone the test. Also I find out another variant, I can change changeDetectionStrategy for testcomponent for Default like this: ```
beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [  ... ]
    })
    .overrideComponent(CourseItemComponent, {  set: {changeDetection: ChangeDetectionStrategy.Default}
    })
    .compileComponents(); 
```
But I think it is a rude decision).